File: Ruby/casilla.rb
require_relative 'estados_juego'
module Civitas
  
  attr_reader :nombre  
  
  class Casilla
    
    @@carcel = 10
    private
    def initialize(_nombre = "")
      @nombre = _nombre
    end
    
    def informe(actual, todos)
      Diario.instance.ocurre_evento("El jugador " + todos[actual].to_s + 
          " ha caido en la casilla" + todos[actual].numCasillaActual.to_s + "\n" + to_s)
    end
    
    def jugador_correcto(actual, todos)
      actual >= 0 && actual < todos.size()
    end
    
    public
    def recibe_jugador(actual, todos)
      informe(actual, todos);
    end
    
    def to_s
      "Casilla de nombre " + @nombre.to_s
    end
  end
end

File: Java/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ilzk
 */
public class Casilla {

    //Atributos de la clase
    static private int carcel;
    private String nombre = "";
    private float importe = 0;
    //Relaciones con las demas clases
    TituloPropiedad tituloPropiedad;
    MazoSorpresas mazo;
    Sorpresa sorpresa;

    /**
     * Metodo que inicia los argumentos
     *
     */
    void init() {
        tituloPropiedad = null;
        sorpresa = null;
        mazo = null;
    }

    /**
     * Constructor de casillas tipo descanso
     *
     * @param _nombre
     */
    Casilla(String _nombre) {
        init();
        nombre = _nombre;
    }

    /**
     * Consultor del nombre de la casilla
     *
     * @return atributo nombre
     */
    public String getNombre() {
        return nombre;
    }

    /**
     * Informa al diario de que ha caido un jugador en la casilla y da los datos
     * de la misma
     *
     * @param actual
     * @param todos
     */
    void informe(int actual, ArrayList<Jugador> todos) {
        Diario.getInstance().ocurreEvento("El jugador " + todos.get(actual).getNombre() + " ha caido en la casilla de caracteristicas " + toString());
    }

    /**
     * Comprueba que existe un jugador en todos en la posicion actual
     *
     * @param actual
     * @param todos
     * @return
     */
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos) {
        return (actual >= 0 && actual < todos.size());
    }

    /**
     * Llama al metodo recibe Jugador indicado para la casilla actual
     *
     * @param actual indice del jugador actual en el vector
     * @param todos vector con todos los jugadores
     */
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        informe(actual, todos);
    }

    @Override
    /**
     * Sobreescritura del metodo toString
     *
     * @return tipo de la casilla, nombre e importe
     */
    public String toString() {
        String devolver = "Casilla de nombre " + nombre;
        return devolver;
    }
}
